docs/manual/constant-value-checker.tex

```diff
@@ -334,6 +334,27 @@
 that possibility, or verify the lack of overflow using some other tool or
 manual analysis.
 
+\sectionAndLabel{Strings can be null in concatenations}{null-strings-concats}
+
+The Constant Value Checker considers all elements of a String concatenation
+non-null by default. The \code{-AnullStringsConcatenation} command-line option
+makes the checker more sound by analyzing the possibility of each element to
+be null. Only String literals, primitive types and effectively final variables
+are considered non-null. Figure~\ref{fig-null-strings-concat} gives an example
+of behavior with and without the \code{-AnullStringsConcatenation} option.
+
+\begin{figure}
+\begin{Verbatim}
+  ...
+  // This String can be null
+  @StringVal("a") String string1;
```

Comment:
`string1` needs to be initialized to something; otherwise, the compiler will reject the next line.  Maybe just `getString()`?